checker.c

```diff
@@ -1,21 +1,68 @@
-#include <stdio.h>
-#include <assert.h>
-
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
-}
+#include <stdio.h>
+#include <string.h>
+#include "checker.h"
+#include "language.h"
+
+
+void Alerter(batteryParam_t param, float reading)
+{
+	printf(alertMsgs[language], param.name, reading, param.lowThreshold, param.hiThreshold);
+}
+
+void WarningCheck(batteryParam_t param, float reading)

```

Comment:
@israel-github there are multiple ways to reduce complexity here.
- Easy way would be to split the function. E.g., lines 16-19 go into `printOnLowWarning` and the rest of it goes to `printOnHighWarning`
- Better way would be to make a function that accepts the limits and prints if it is within those limits
- Even better way would be to make an array of structures, each having a value and a classification. Then make a for loop that will go through them, assigning the classification whenever the input is within that value. Separating data from code. That way we can introduce more ranges without increasing the complexity.